astropy/constants/CODATA2010.py

```diff
@@ -0,0 +1,125 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Astronomical and physics constants in SI units.  See :mod:`astropy.constants`
+for a complete listing of constants defined in Astropy.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+
+from .constant import Constant
+
+
+# PHYSICAL CONSTANTS
+
+class Codata2010(Constant):
+    default_reference = 'CODATA 2010'
+    _registry = {}
+    _has_incompatible_units = set()
+
+    def __new__(cls, abbrev, name, value, unit, uncertainty,
```

Comment:
I think rather than overwriting `__new__`, I would slightly adjust `Constant.__new__`, setting `reference=None` in the keyword list, and if it is `None`, changing it to `cls.default_reference` if that exists, i.e.,
```
if reference is None:
    reference = getattr(cls, 'default_reference', None)
    if reference is None:
        raise TypeError("{} requires a reference.".format(cls))
```